### PR TITLE
Adds chunk struct and its generator

### PR DESCRIPTION
**Pull request by cuducos**
This PR suggests a simple structure to start handling our chunks, as described in #4. We can start using it once #10 is in.

### CODE REVIEW

File: main_test.go [hunk 1]
```diff
@@ -176,3 +176,28 @@ func TestDownloadWithContext_ErrorUserTimeout(t *testing.T) {
 		t.Error("expected channel closed, but did not get it")
 	}
 }
+
+func TestDownload_Chunks(t *testing.T) {
+	d := DefaultDownloader()
```
Comment (danielfireman) [inline main_test.go:181]: How about explicitly setting the `ChunkSize`?

**Review by danielfireman — APPROVED** [review]

File: main_test.go (hunk 1 above)
Comment (cuducos) [reply main_test.go:181]: Good. Then we can make less use of “magic numbers” and work with numbers our brain can parse easily (12 total bytes, chunk size of 5 etc).

Comment (cuducos) [reply main_test.go:181]: Bem melhor 87d92d9dca4fdfc41a68533f95bf0cb064358bf2 ✨

Comment (danielfireman) [reply main_test.go:181]: > Good. Then we can make less use of “magic numbers” and work with numbers our brain can parse easily (12 total bytes, chunk size of 5 etc).

Exactly! 😄

**Pull request merged**